src/opencl_sha512crypt_fmt_plug.c

```diff
@@ -305,15 +305,9 @@ static void clear_keys(void)
 
 static void set_key(char *key, int index)
 {
-	int len;
-
-	//Assure buffer has no "trash data".
-	memset(plaintext[index].pass, '\0', PLAINTEXT_LENGTH);
-	len = strlen(key);
-	len = (len > PLAINTEXT_LENGTH ? PLAINTEXT_LENGTH : len);
```

Comment:
This is not required in jumbo (@magnumripper?). 

I removed it from raw formats, because they are more relevant there. Ok, I know you will revert that (un)requirement some day.

Reply:
If you mean the `len > PLAINTEXT_LENGTH`, yes - Jumbo is supposed to never ever serve an overlong key. There has been bugs though, hopefully none are left. The original idea for it came from very fast CPU formats (eg. NT) where `set_key()` (and key generation) is the bottleneck.